cmd/readiness/main.go

```diff
@@ -45,59 +44,38 @@ func init() {
 // Additionally if the previous check hasn't returned 'true' the "deadlock" case is checked to make sure the Agent is
 // not waiting for the other members.
 func isPodReady(conf config.Config) bool {
-	fd, err := os.Open(conf.HealthStatusFilePath)
+	healthStatus, err := conf.HealthStatusProvider.HealthStatus()
 	if err != nil {
-		logger.Warn("No health status file exists, assuming the Automation agent is old")
```

Comment:
note, that this behavior is not true anymore as the absence of the health file will be considered as an error and the pod won't be marked as "ready".
This condition here was true a couple of years ago when the support for health file was added (middle of August 2019: https://github.com/10gen/mms-automation/pull/2307) to support "old" agents.
This refers to some 3.6/4.0 version of OM which are [not supported](https://docs.google.com/spreadsheets/d/1nTvZ2cTg2Ef2XweyV1ZBb7BMSYO4eKqXszfqkpBsOPM/edit#gid=0) anymore.

So it should be safe to consider that the version of the Agent used will always eventually write the health file (usually it happens in a couple of seconds after the Agent starts).